documentation/general/design/dash-holistic-view.md

```diff
@@ -0,0 +1,627 @@
+Technology suppliers must implement this interface for their DASH devices. This
+is the primary integration point of DASH devices and the SONiC stack. It will be
+rigorously tested for performance and conformance. See [DASH Testing
+documentation](https://github.com/Azure/DASH/tree/main/test).
+
+SAI "schema" are represented as fixed c-language header files and derived
+metadata header files. The underlay and overlay schema have different origins:
+
+- Traditional SAI headers are defined in the [OCP SAI project
+  repo](https://github.com/opencomputeproject/SAI/tree/master/inc).These are
+  hand-generated and maintained. DASH uses a subset of these to manage "underlay"
+  functions, e.g. device management, Layer 3 routing and so forth.
+- DASH SAI "overlay" objects are derived from a [P4 Behavioral
+  Model](https://github.com/Azure/DASH/tree/main/sirius-pipeline). A script
+  reads the P4 model and generates SAI header files.
+
+DASH uses an **enhanced syncd** to configure the dataplane using the technology
+supplier-specific SAI library.
+
+### ASIC Drivers
+
+The term "ASIC Drivers" is borrowed from traditional SONiC and SAI, where a
+datacenter switch was implemented almost entirely inside an ASIC (or multiple
+ASICs). These devices are programmed using a technology supplier Software
+Development Kit (SDK) which includes device drivers, kernel modules, etc.
+
+A contemporary DASH "SmartNIC" may consist of many complex hardware components
+including multi-core System On A Chip (SoC) ASICs, and the associated software.
+For simplicity, the software for such systems which interfaces to the SAI layer
+is collectively called the "ASIC driver." More importantly, the technology
+supplier SAI library will hide all details and present a uniform interface.
+
+
+> [!NOTE]
+> A dedicated group has been investigating the use of PINS.  
+
+**Why DASH and Why Not PINS?**  PINS is based upon a specific language (P4), and
+DPUs on the market would need to build a cross compiler. PINS places the onus
+onto the Cloud Providers to code and string together the PINS API and their own
+implementation to create the actions they want to achieve; however it may not be
+optimal for performance reasons in our project.
+
+### DASH capable ASICs
+
+These comprise the main dataplane engines and are the core of what are variously
+called SmartNICs, DPUs, IPUs, NPUS, etc. The actual cores may be ASICs, SoCs,
+FPGAs, or some other high-density, performant hardware.
+
+## SONiC integration
+
+The system architecture for SONiC DASH relies upon the [SONiC system
+architecture](https://github.com/Azure/SONiC/wiki/Architecture). For more
+information and details about the integration, see [SONiC DASH
+HLD](https://github.com/Azure/DASH/blob/main/documentation/general/design/dash-sonic-hld.md). 
+
+
+The SONiC DASH integration introduces the following DASH modifications:
+
+1. A *new docker container* in the user space named **DASH container** (aka SDN
```

Comment:
this would be mostly named gNMI container.

Reply:
Fixed